legate/core/operation.py

```diff
@@ -399,13 +397,39 @@ def add_scalar_arg(
         ----------
         value : Any
             Scalar value or a tuple of scalars (but no nested tuples)
-        dtype : DType
+        dtype : Dtype
             Data type descriptor for the scalar value. A descriptor ``(T,)``
-            means that the value is a tuple of elements of type ``T``.
+            means that the value is a tuple of elements of type ``T`` (i.e.,
+            equivalent to ``array_type(T, len(value))``).
         """
-        if not isinstance(dtype, (ty.Dtype, tuple)):
-            raise ValueError(f"Unsupported type: {dtype}")
-        self._scalar_args.append((value, dtype))
+        sanitized: ty.Dtype
+        if isinstance(dtype, ty.Dtype):
+            sanitized = dtype
+            if isinstance(sanitized, ty.FixedArrayDtype) and not isinstance(
+                value, (tuple, Shape)
+            ):
+                raise ValueError(f"{value} is not a valid scalar")
```

Comment:
Should we also check the length of `value` against `sanitized. num_elements()` at this point?

Reply:
https://github.com/nv-legate/legate.core/pull/785/commits/119261a765d2ae3585b4053fa5ee12c9ef6c3f28